modules/puzzle/src/main/PuzzleActivity.scala

```diff
@@ -21,21 +21,25 @@ final class PuzzleActivity(
 
   def stream(config: Config): Source[JsObject, ?] =
     val perSecond = MaxPerSecond(20)
-    Source.futureSource:
-      colls.round.map:
-        _.find(
-          $doc(PuzzleRound.BSONFields.user -> config.user.id) ++
-            config.before.so: before =>
-              $doc(PuzzleRound.BSONFields.date.$lt(before))
-        )
-          .sort($sort.desc(PuzzleRound.BSONFields.date))
-          .batchSize(perSecond.value)
-          .cursor[PuzzleRound](ReadPref.sec)
-          .documentSource(config.max | Int.MaxValue)
-          .grouped(perSecond.value)
-          .throttle(1, 1 second)
-          .mapAsync(1)(enrich(config))
-          .mapConcat(identity)
+    val baseQuery = $doc(PuzzleRound.BSONFields.user -> config.user.id)
+    val timeQueries = List(
+      config.before.map(before => $doc(PuzzleRound.BSONFields.date.$lt(before))),
+      config.since.map(since => $doc(PuzzleRound.BSONFields.date.$gte(since)))
+    )
```

Comment:
```suggestion
    ).flatten
    val finalQuery = baseQuery ++ timeQueries
```

Reply:
or maybe: 
```scala
    val queries = $doc(PuzzleRound.BSONFields.user -> config.user.id) ++ List(
      config.before.map(before => $doc(PuzzleRound.BSONFields.date.$lt(before))),
      config.since.map(since => $doc(PuzzleRound.BSONFields.date.$gte(since)))
    ).flatten
```

Reply:
With this approach I am getting `None of the overloaded alternatives of method ++ in class BSONDocument with types...` 🤔

Reply:
mm, maybe it was an oversight from me. I'll check it real quick.

Reply:
So, We can do like this: `val finalQuery = baseQuery.++(timeQueries*)`, ugly but I think it's a bit more performance as We create less allocation.

Reply:
Alright, dbbbc13 should take care of that. Thanks for the help - first time working with Scala 😅.

Reply:
you did really well, hope to see more prs from you!